Jetpack Cloud Cart Integration - Fix mobile cart alignment

vikneshwar opened the pull request:
#### Proposed Changes

 - Fix mobile banner cart alignment
 - Convert displayName from component to plain string to be shown in JITM
 - Calculate the position of the cart in the main banner and use the same for the banner transition (on desktop)
 - When no item is in the cart hide the count instead of not displaying it

#### Testing Instructions

* Make sure you are logged-in to WordPress.com 
* click on Jetpack Cloud live link below
* or boot up this PR 
    * Run `git fetch && git checkout fix/cloud-cart-lightbox-cta`
    * Run `yarn start-jetpack-cloud`
    * Goto http://jetpack.cloud.localhost:3000
* you will be presented with a landing page, asking you to select a site, choose any site from the list
* Once selected, you will be redirected to `/backup/:site-slug` page by default, replace `backup` with `pricing` and you should land in pricing page
* Now append this query string (`?flags=jetpack/pricing-page-cart`) to the end of URL to enable the cart and reload the page
* Make sure the cart shows up in the header
* Click on Add to Cart on Akismet Anti-Spam and make sure JITM says **Akismet Anti-Spam added to cart**
* Now switch to mobile view and confirm the cart is displayed just before the menu 
* Now slowly scroll down to *14 days money back guarantee* banner and confirm that the banner becomes sticky and cart is shown in the banner and Jetpack logo masterbar banner hides (*checkout the screencast*)
* Also validate that cart is positioned in the same location when moving between banners in different screen size (***refer screencast 2 for different scenarios to be tested. The cart will transition to the same location between banners only for view >660px***)

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->
#### Screencast


https://user-images.githubusercontent.com/2027003/210622743-a3420978-4594-444e-acd6-d02ca78b6388.mov


##### Cart transition between banners for different screen sizes



https://user-images.githubusercontent.com/2027003/210809639-5fa49f5b-d668-417a-b131-d09f42dabb78.mov



#### Screenshot

| Before | After |
|--------|------|
| <img width="500" alt="Screen Shot 2022-12-06 at 4 33 54 PM" src="https://user-images.githubusercontent.com/2027003/210623393-7bf499a6-fa08-4243-bad9-ecd7e6e0441a.png"> | <img width="500" alt="Screen Shot 2022-12-06 at 3 26 40 PM" src="https://user-images.githubusercontent.com/2027003/210623366-f8639935-d4be-458d-b62c-4fe825ea20ee.png"> |


#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

vikneshwar committed:
Jetpack Pricing Page Cart Integration - fix styles
 - Fix mobile banner cart alignment
 - Convert displayName from component to plain string to be shown in JITM

vikneshwar committed:
Jetpack cloud cart integration - fix stylelint error

andrii-lysenko reviewed (commented):
It already looks great! Eager to see the final result 👍

vikneshwar committed:
Jetpack Cloud Cart Integration - fix cart banner position
- Calculate the position of the cart in main banner and use the same for the banner trasition (in desktop)
- When no item in cart hide the count instead of not displaying it

andrii-lysenko commented on client/components/jetpack/intro-pricing-banner/index.tsx:
```diff
@@ -16,9 +17,40 @@ import people from './people.svg';
 const CALYPSO_MASTERBAR_HEIGHT = 47;
 const CLOUD_MASTERBAR_HEIGHT = 0;
 
+const useBoundingClientRect = ( eleQuery: string ) => {
```
Maybe we could move this hook into a separate file to have it cleaner.

andrii-lysenko reviewed (approved):
LGTM! A small suggestion to keep things cleaner.

andrii-lysenko committed:
update: move hook into separate file and fix styles
